frontend/src/main/scala/bloop/dap/DebuggeeRunner.scala

```diff
@@ -0,0 +1,51 @@
+package bloop.dap
+
+import bloop.data.{Platform, Project}
+import bloop.engine.State
+import bloop.engine.tasks.{RunMode, Tasks}
+import bloop.exec.JavaEnv
+import ch.epfl.scala.bsp.ScalaMainClass
+import monix.eval.Task
+
+trait DebuggeeRunner {
+  def run(logger: DebugSessionLogger): Task[Unit]
+}
+
+private final class MainClassDebugAdapter(
+    project: Project,
+    mainClass: ScalaMainClass,
+    env: JavaEnv,
+    state0: State
+) extends DebuggeeRunner {
+  def run(logger: DebugSessionLogger): Task[Unit] = {
+    val stateForDebug = state0.copy(logger = logger)
+    val workingDir = state0.commonOptions.workingPath
+    val runState = Tasks.runJVM(
+      stateForDebug,
```

Comment:
Let's rename `state0` to `state` and use `state` here, as `stateForDebug == state0`.

Reply:
We must inject our logger, so those states are not equal

Reply:
Good catch, I misread this! Can we do instead `state0.copy(logger = debugLogger)`? I thought `logger` was `state0.logger` and that it was in scope, as that's usually the case in all of our implementation, so that's what confused me.